packages/terra-application-header-layout/src/ApplicationHeaderLayout.jsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import AppDelegate from 'terra-app-delegate';
+
+import styles from './ApplicationHeaderLayout.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The AppDelegate instance provided by the containing component. If present, its properties will propagate to the children components.
+   * */
+  app: AppDelegate.propType,
+  /**
+   * Extensions element to be placed before the end of the header.
+   * */
+  extensions: PropTypes.element,
+  /**
+   * Logo element to be placed at the start of the header after the toggle element.
+   * */
+  logo: PropTypes.element,
+  /**
+   * Layout configuration to handle size and toggle functionality.
+   * */
+  layoutConfig: PropTypes.shape({
+    size: PropTypes.string,
+    toggleMenu: PropTypes.func,
+    menuIsOpen: PropTypes.bool,
+  }),
+  /**
+   * Navigation element to be placed within the fill area of the header.
+   * */
+  navigation: PropTypes.element,
+  /**
+   * Toggle element to be placed at the start of the header.
+   * */
+  toggle: PropTypes.element,
+  /**
+   * Utilities element to be placed at the end of the header.
+   * */
+  utilities: PropTypes.element,
+};
+
+const defaultProps = {
+  layoutConfig: {
+    menuIsOpen: false,
+  },
+};
+
+const appendSizeToElement = (app, size, element) => React.cloneElement(element, { app, size });
+const appendConfigToElement = (app, layoutConfig, element) => React.cloneElement(element, { app, layoutConfig });
+
+const ApplicationHeaderLayout = ({
+  app,
+  extensions,
+  layoutConfig,
+  logo,
+  navigation,
+  toggle,
+  utilities,
+  ...customProps
+}) => {
+  const headerClassNames = cx([
+    'header',
+    'fill',
+    customProps.className,
+  ]);
+
+  let logoElement;
+  if (logo) {
+    const clonedElement = appendSizeToElement(app, layoutConfig.size, logo);
+    logoElement = <div className={cx(['fit', 'start', 'logo'])}>{clonedElement}</div>;
+  }
+
+  let navigationElement;
+  if (navigation) {
+    const clonedElement = appendSizeToElement(app, layoutConfig.size, navigation);
+    navigationElement = <div className={cx('fill', 'navigation')}>{clonedElement}</div>;
```

Comment:
`navigation` won't be scoped since it has no styling. It should be dropped or added as a `data-terra-framework-app-header-layout-navigation`

Reply:
same with extensions and header-toggle